lib/Entities/NewsletterEntity.php

```diff
@@ -310,7 +310,7 @@ function setUnsubscribeToken($unsubscribe_token) {
    * @return NewsletterEntity|null
    */
   function getParent() {
-    return $this->parent;
+    return empty($this->parent_id) ? null : $this->parent;
```

Comment:
It seems that `empty($this->parent_id)` will always evaluate as true even if the entity has set existing parent so after this change, `getParent` will always return `null`.

I think we should be catching EntityNotFoundException exception. It could be done like that:
```
try {
      if ($this->parent && $this->parent->getId()) {
        return $this->parent;
      }
    } catch (EntityNotFoundException $enf) {
      $this->setParent(null);
      return null;
    }
```